DQMOffline/MuonDPG/interface/BaseTnPEfficiencyTask.h

```diff
@@ -0,0 +1,108 @@
+#ifndef BaseTnPEfficiencyTask_H
+#define BaseTnPEfficiencyTask_H
+
+/*
+ * \file BaseTnPEfficiencyTask.h
+ *
+ * \author L. Lunerti - INFN Bologna
+ *
+*/
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include "CommonTools/Utils/interface/AnySelector.h"
+
+#include "DataFormats/DTRecHit/interface/DTRecSegment4D.h"
+#include "DataFormats/DTRecHit/interface/DTRecSegment4DCollection.h"
+
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+
+#include <vector>
+#include <string>
+#include <map>
+
+class BaseTnPEfficiencyTask : public DQMEDAnalyzer {
+public:
+  /// Constructor
+  BaseTnPEfficiencyTask(const edm::ParameterSet& config);
+
+  /// Destructor
+  ~BaseTnPEfficiencyTask() override;
+
+protected:
+  /// BeginRun
+  void dqmBeginRun(const edm::Run& run, const edm::EventSetup& context) override;  //final ?
+
+  void bookHistograms(DQMStore::IBooker& iBooker, edm::Run const& run, edm::EventSetup const& context) override;
+
+  /// Return the top folder
+  virtual std::string topFolder() const = 0;
+
+  /// Analyze
+  void analyze(const edm::Event& event, const edm::EventSetup& context) override;  // final?
+  bool hasTrigger(std::vector<int>& trigIndices,
+                  const trigger::TriggerObjectCollection& trigObjs,
+                  edm::Handle<trigger::TriggerEvent>& trigEvent,
+                  const reco::Muon& muon);
+
+  std::vector<std::vector<unsigned>> m_probeIndices;
+  std::vector<std::vector<unsigned>> m_tagIndices;
+
+  std::map<std::string, MonitorElement*> m_histos;
+
+  int m_nEvents;
+
+  edm::EDGetTokenT<reco::MuonCollection> m_muToken;
+
+  double m_borderCut;
```

Comment:
Also all these parameters can be declared `const`, as well as several other ones in the private members below

Reply:
I'm adding `const` wherever possible.